validation/poco-validation.rst

```diff
@@ -0,0 +1,110 @@
+   * - CodePatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``code``.
+   * - DatePatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``date``.
+   * - DateTimePatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``dateTime``.
+   * - IdPatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``id``.
+   * - InstantPatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``id``.
+   * - NarrativeXhtmlPattern
+     - Verifies whether the string property (normally inside Div) in ``Div`` is valid XML and (if configured to do so), validates whether this XML adheres to the `additional rules for FHIR narrative <https://www.hl7.org/fhir/narrative.html>`_.
+   * - OidPatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``oid``.
+   * - TimePatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``time``.
+   * - UriPatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``uri``.
+   * - UuidPatternAttribute
+     - Verifies whether the string property is a correctly formatted FHIR ``uuid``.
+
+In addition, the POCO classes implement `IValidatableObject <https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.ivalidatableobject>`_ to perform additional validations not bound to a single property:
+
+.. list-table:: Additional non-attribute validations
+   :widths: 10 90
+   :header-rows: 1
+
+   * - POCO
+     - Validation performed
+   * - *All resources*
+     - Validates that contained resources do not themselves have nested contained resources
+   * - Code<T>
+     - Validates that the string in ``ObjectValue`` can be parsed as one of the enum values in the enumeration ``T``.
+
+As is clear from the set of validations described above, the attribute-based validation does not cover all constraints that can be expressed using a ``StructureDefinition``, not even all those for the non-profiled FHIR core datatypes. Notable constrains *not* checked are:
+
+* **Terminology** - the attribute validator will not call upon a terminology service to validate the coded datatypes, it will however validate correct values for all required, explicit bindings, since these are generated as .NET enumerations.
+* **Slices** - these are not used in the definitions of the core resources from which the POCOs are generated. There is also no trivial method to express these constraints in a .NET class model, so attribute validation does not cover them.
+* **FhirPath invariants** - FHIR model definitions generally contain additional invariants specified using `FhirPath <http://hl7.org/fhirpath/>`_. These are not generated into the POCO classes and require an external FhirPath engine.
+
```

Comment:
Perhaps we should also mention that we have a full validator available, for these kind of use cases. For example with a link to this documentation